src/CanCommunity.ts

```diff
@@ -224,23 +224,23 @@ export class CanCommunity {
   async checkRightHolder(rightHolder: RightHolder, account: EosName) {
     // check right holder is set or not
     const isSetRightHolder =
-      rightHolder.accounts.length !== 0 ||
-      rightHolder.required_badges.length !== 0 ||
-      rightHolder.required_positions.length !== 0 ||
-      rightHolder.required_tokens.length !== 0;
+      rightHolder.accounts.length > 0 ||
+      rightHolder.required_badges.length > 0 ||
+      rightHolder.required_positions.length > 0 ||
+      rightHolder.required_tokens.length > 0;
 
     if (!isSetRightHolder) {
       // return false if right holder is not set
       return false;
     }
 
     // check user account satisfy require accounts,
-    if (rightHolder.accounts.length !== 0 && !rightHolder.accounts.includes(account)) {
-      return false;
+    if (rightHolder.accounts.includes(account)) {
```

Comment:
should be `rightHolder.accounts.length > 0 && rightHolder.accounts.includes(account)`